### ISSUE TITLE
Wrong language during login.

### ISSUE THREAD

**Issue by Sama34** [mybb/mybb#2059]
Default board language: Spanish.
User Language: English

When login-in correctly the redirection page is shown on the board default language but it should be shown with the user language.

**Comment by Stefan-ST** [comment]
Technically the user isn't logged in at that stage. The same applies probably for basically every other user settings, too.

**Comment by JN-Jones** [comment]
From a user perspective I'd say that the redirection page should be in the user language. As the language string is loaded after the user credentials are checked it should be possible to load his language for the redirection page. Other user settings shouldn't matter at this point. @Stefan-ST

**Comment by Stefan-ST** [comment]
"Show friendly redirect pages" and the theme setting might matter here if you want a consistent behaviour.

**Comment by euantorano** [comment]
Yes, I agree with Stefan. If I login and have friendly redirect pages turned off, I'd expect to not see a friendly redirect.

**Comment by JN-Jones** [comment]
That setting shouldn't be a problem to check too. However the theme might be an issue as it's usually loaded in global.php afaik. Guess someone would need to check whether it's possible to properly change the theme after the user has been loaded.

**Comment by euantorano** [comment]
The theme isn't as big of a problem IMO if it's going to require a big change - if friendly redirects are disabled, it won't even be seen either.

**Comment by Ben-MyBB** [comment]
@euantorano happy if this one is just closed given it's age?

**Comment by euantorano** [comment]
I’d leave it open, it would be very nice to get this fixed at some point. It’s fairly minor, but is an annoyance.

**Comment by rajat315315** [comment]
are we talking to disable/remove "friendly re-direct pages" option permanently? or should we try to load user lang before the redirect happens?

**Comment by euantorano** [comment]
@rajat315315 There are 2 issues at play here really:

- When logging in, the redirection page should use the user's language, even on failure.
- When logging in, the redirection page shouldn't show if users have them disabled

In the future, I can see us removing the redirection pages completely and having redirection notices when needed.

**Comment by Sama34** [comment]
> even on failure.

Seems a little too much, because you are disclosing 1) the user exists and 2) the user language setting.
